plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/rules/model/CorrelationRule.java

```diff
@@ -0,0 +1,239 @@
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static org.opensearch.plugin.correlation.utils.CorrelationConstants.NO_ID;
+import static org.opensearch.plugin.correlation.utils.CorrelationConstants.NO_VERSION;
+
+/**
+ * Correlation Rule DSL
+ * {
+ *   "id": "",
+ *   "version": 1,
+ *   "name": "s3 to app logs",
+ *   "correlate": [
+ *     {
+ *       "index": "s3_access_logs",
+ *       "query": "aws.cloudtrail.eventName:ReplicateObject",
+ *       "timestampField": "@timestamp",
+ *       "tags": [
+ *         "s3"
+ *       ]
+ *     }
+ *   ]
+ * }
+ *
+ * @opensearch.api
+ */
+public class CorrelationRule implements Writeable, ToXContentObject {
+
+    private static final Logger log = LogManager.getLogger(CorrelationRule.class);
+
+    /**
+     * Correlation Rule Index
+     */
+    public static final String CORRELATION_RULE_INDEX = ".opensearch-correlation-rules-config";
+
+    private static final ParseField ID_FIELD = new ParseField("id");
+    private static final ParseField VERSION_FIELD = new ParseField("version");
+    private static final ParseField NAME_FIELD = new ParseField("name");
+    private static final ParseField CORRELATION_QUERIES_FIELD = new ParseField("correlate");
+    private static final ObjectParser<CorrelationRule, Void> PARSER = new ObjectParser<>("CorrelationRule", CorrelationRule::new);
+
+    static {
+        PARSER.declareString(CorrelationRule::setId, ID_FIELD);
+        PARSER.declareLong(CorrelationRule::setVersion, VERSION_FIELD);
+        PARSER.declareString(CorrelationRule::setName, NAME_FIELD);
+        PARSER.declareField((xcp, rule, context) -> {
+            List<CorrelationQuery> correlationQueries = new ArrayList<>();
+            XContentParser.Token currentToken = xcp.currentToken();
+            if (currentToken == XContentParser.Token.START_ARRAY) {
+                while (xcp.nextToken() != XContentParser.Token.END_ARRAY) {
+                    correlationQueries.add(CorrelationQuery.parse(xcp));
+                }
+            }
+            rule.setCorrelationQueries(correlationQueries);
+        }, CORRELATION_QUERIES_FIELD, ObjectParser.ValueType.OBJECT_ARRAY);
+    }
+
+    private String id;
+
+    private Long version;
+
+    private String name;
+
+    private List<CorrelationQuery> correlationQueries;
+
+    private CorrelationRule() {}
+
+    /**
+     * Parameterized ctor of Correlation Rule
+     * @param id id of rule
+     * @param version version of rule
+     * @param name name of rule
+     * @param correlationQueries list of correlation queries part of rule
+     */
+    public CorrelationRule(String id, Long version, String name, List<CorrelationQuery> correlationQueries) {
+        this.id = id != null ? id : NO_ID;
```

Comment:
can we just have another ctor that doesn't take an ID as a parameter?

Reply:
added a new ctor.